### ISSUE TITLE
Passing type `never` wrongly as argument to function does not throw error

### ISSUE THREAD

**Issue by chialuka** [microsoft/TypeScript#49714]
# Bug Report

### 🔎 Search Terms
passing never as argument of a function

### 🕗 Version & Regression Information
4.6.2

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about ___type never as function argument___

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYewdgzgLgBMCGAbRBhAFgU2AawwJxgF4YAKASiID4YBvAKBjnGjkxyJgHJNkROBuBjACWAM1JQAngAcMIccDbYihYpzABXALYAjfJwr1GjRVlx4AYhrDAow8CRowAJvEkAuVmYA0MO1oxPU3YAXzJBRhC6EME6UEhYYPMrGztwDkcXN19-DBgQzydXDxgdEBBEDHgwfj9hAM8yiqqwfIpCaiMYPAwoDTxWouy6gPzBGKA)

### 💻 Code

```ts
const callChecker = () => {
  const check = 'hello';
  if (typeof check === 'number') {
   // check is of type `never` here
    checkerFunction({ day: check, time: check });
  }
};

const checkerFunction = ({ day, time }: { day: boolean; time: boolean }) => {
  return { day, time };
};
```

### 🙁 Actual behavior
TypeScript does not throw an error although I am trying to assign type `never` to type `boolean`

### 🙂 Expected behavior

I expected I'd get an error showing the types were incompatible

**Comment by MartinJohns** [comment]
That's working as intended. Nothing is assignable to `never`, but `never` is assignable to anything.

**Comment by chialuka** [comment]
Thanks for clarifying @MartinJohns!
